### ISSUE TITLE
different type of imported sweetalert2  

### ISSUE THREAD

**Issue by sup-ham** [PepsRyuu/nollup#203]
different type of imported sweetalert2 in nollup (top) vs rollup (bottom)

code 
```
import Swal from 'sweetalert2';
console.info({Swal})
```

![image](https://user-images.githubusercontent.com/3823496/121709491-d17dfc80-cb02-11eb-89d6-aeca7fd30264.png)

**Comment by PepsRyuu** [comment]
Can I see your ```package.json``` and ```rollup.config.js```?

**Comment by sup-ham** [comment]
package.json
```
{
  "name": "svelte-app",
  "version": "1.0.0",
  "@comments scripts": {
    "dev": "develop with blazing fast rebuilds",
    "dev:features": "develop with features like SSR and serviceworker enabled",
    "build": "run build scripts below",
    "build:app": "build single page application (SPA)",
    "build:static": "Generate static pages",
    "serve": "serve content in 'dist' folder",
    "rollup": "run the rollup bundler",
    "nollup": "run the nollup no-bundler",
    "routify": "run routify"
  },
  "scripts": {
    "dev": "run-p routify rollup",
    "devn": "run-p routify nollup",
    "build": "run-s build:app",
    "build:app": "routify -b && rollup -c configs/rollup.js",
    "build:static": "spank",
    "serve": "spassr --ssr",
    "rollup": "rollup -c configs/rollup.js -w",
    "nollup": "nollup --rc configs/nolluprc.js",
    "routify": "routify"
  },
  "devDependencies": {
    "@rollup/plugin-commonjs": "^19.0.0",
    "@rollup/plugin-node-resolve": "^13.0.0",
    "@roxi/routify": "^2.18.0",
    "node-phpcgi": "github:sup-ham/node-phpcgi#main",
    "nollup": "^0.16.5",
    "npm-run-all": "^4.1.5",
    "postcss": "^8.3.2",
    "postcss-import": "^14.0.2",
    "rollup": "^2.51.2",
    "rollup-plugin-hot": "^0.1.1",
    "rollup-plugin-livereload": "^2.0.0",
    "rollup-plugin-postcss": "^4.0.0",
    "rollup-plugin-svelte-hot": "^1.0.0-8",
    "rollup-plugin-terser": "^7.0.2",
    "rollup-plugin-workbox": "^6.1.1",
    "spank": "^1.8.0",
    "spassr": "^2.6.0",
    "svelte": "^3.38.2",
    "svelte-flatpickr": "^3.1.1",
    "svelte-imask": "^1.2.0",
    "svelte-paginate": "^0.0.1",
    "svelte-preprocess": "^4.7.3",
    "sweetalert2": "^11.0.16"
  },
  "routify": {
    "extensions": "svelte,html,svx,md"
  },
  "spassr": {},
  "spank": {
    "blacklist": [
      "/example/modal/basic/4"
    ]
  }
}
```

rollup config
```
import postcss from 'rollup-plugin-postcss';
import postcssImport from 'postcss-import';
import svelteRollup from 'rollup-plugin-svelte-hot';
import Hmr from 'rollup-plugin-hot';
import resolve from '@rollup/plugin-node-resolve';
import commonjs from '@rollup/plugin-commonjs';
import livereload from 'rollup-plugin-livereload';
import getConfig from '@roxi/routify/lib/utils/config';
import { spassr } from 'spassr';
import { terser } from 'rollup-plugin-terser';
import { copySync, removeSync, writeFileSync } from 'fs-extra';
import { injectManifest } from 'rollup-plugin-workbox';

const { server, globals, svelte } = getConfig();
const { distDir, spaEntry, assetsDir, getTemplate, apiMiddleware } = server;
const buildDir = `${distDir}/build`;
const isNollup = !!process.env.NOLLUP;
const production = !process.env.ROLLUP_WATCH;

// clear previous builds
production && (removeSync(distDir) || removeSync(buildDir));

const serve = _ => ({
    writeBundle: async _ => {
        const options = {
            assetsDir: [assetsDir, distDir],
            entrypoint: `${assetsDir}/${spaEntry}`,
            script: `${buildDir}/main.js`,
        }
        spassr({ ...options, port: 5000, middleware: apiMiddleware });
        // spassr({ ...options, ssr: true, port: 5005, ssrOptions: { inlineDynamicImports: true, dev: true } })
    }
})

const copyToDist = _ => ({
    writeBundle() {
        copySync(assetsDir, distDir);
        writeFileSync(distDir + '/' + spaEntry, getTemplate(), { encoding: "utf8", flag: 'w', mode: 0o666 });
    }
})


export default {
    preserveEntrySignatures: false,
    input: [`src_front/main.js`],
    output: {
        sourcemap: true,
        format: 'esm',
        dir: buildDir,
        // for performance, disabling filename hashing in development
        chunkFileNames: `[name]${production && '-[hash]' || ''}.js`,
    },
    plugins: [
        svelteRollup({
            compilerOptions: {
                dev: !production, // run-time checks
            },
            hot: isNollup,
            ...svelte,
        }),
        // Extract component CSS — better performance
        postcss({
            plugins: [postcssImport()],
            extract: 'bundle.css',
        }),

        // resolve matching modules from current working directory
        resolve({
            browser: true,
            dedupe: importee => !!importee.match(/svelte(\/|$)/),
        }),
        commonjs(),

        production && terser(),
        !production && !isNollup && serve(),
        !production && !isNollup && livereload(distDir), // refresh entire window when code is updated
        !production && isNollup && Hmr({ inMemory: true, public: assetsDir, baseUrl: globals.BASE_URL || '/' }), // refresh only updated code
        injectManifest({
            globDirectory: assetsDir,
            globPatterns: ['**/*.{js,css,svg}', spaEntry],
            swSrc: `src_front/utils/sw.js`,
            swDest: `${distDir}/serviceworker.js`,
            maximumFileSizeToCacheInBytes: 10000000, // 10 MB,
            mode: 'production',
        }),
        production && copyToDist(),
    ],
    watch: {
        clearScreen: false,
        buildDelay: 100,
    },
}
```

additional config read by getConfig()
```
const { existsSync, readFileSync } = require('fs');
const path = require('path');
const autoPreprocess = require('svelte-preprocess');
const express = require('express');

const cfg = module.exports = {
  pages: './src_front/pages',
  phpcgi: {
    env: {},
    docRoot: path.join(__dirname, '/../src_api/web'),
    bin: '/usr/bin/php-cgi',
  },
  globals: { BASE_URL: '', API_URL: '/api' },
  server: {
    distDir: 'app',
    spaEntry: 'index.php',
    assetsDir: 'src_front/assets',
  }
}

const a11y = warn => warn.code.indexOf('a11y') >= 0;
const alowedGlobals = warn => warn.code === 'missing-declaration' && cfg.globals.hasOwnProperty(warn.message.split("'")[1]);

cfg.server.getTemplate = _ => {
  let html = readFileSync(`${cfg.server.assetsDir}/${cfg.server.spaEntry}`, { encoding: 'utf8' });
  let reducer = (prev, [n, v]) => `${prev}    window.${n} = '${v}';\n`;
  let globals = Object.entries(cfg.globals).reduce(reducer, '\n');
  return html.replace('$globals', globals);
}

existsSync(__dirname + '/main.env.js') && require('./main.env')(cfg);
process.env.BASE_URL && (cfg.globals.BASE_URL = process.env.BASE_URL);
process.env.API_URL && (cfg.globals.API_URL = process.env.API_URL);

cfg.server.apiMiddleware = app => {
  const phpcgi = require("node-phpcgi");
  
  cfg.phpcgi.env.CGI_BASE = cfg.globals.API_URL;
  app.use(cfg.globals.API_URL, express.static(cfg.phpcgi.docRoot));
  app.use(cfg.globals.API_URL, phpcgi(cfg.phpcgi));
  app.use((req, res, next) => {
    if (req.method !== 'GET' && req.method !== 'HEAD' && !req.accepts('html')) return next();
    res.send(cfg.server.getTemplate());
  });
}

module.exports.svelte = {
  onwarn(warning, handler) {
    switch (true) {
      case a11y(warning):
      case alowedGlobals(warning):
        return;
    }
    handler(warning);
  },
  preprocess: [
    // custom syntax: https://gist.github.com/sup-ham/73a57c88652a62c26fd23bca99ecf1ed
    autoPreprocess({
      replace: [
        [/{@if /gim, '{#if '],
        [/{@else if /gim, '{:else if '],
        [/{@elseif /gim, '{:else if '],
        [/{@else}/gim, '{:else}'],
        [/{@endif}/gim, '{/if}'],
        [/{@each /gim, '{#each '],
        [/{@foreach /gim, '{#each '],
        [/{@endeach}/gim, '{/each}'],
        [/{@endforeach}/gim, '{/each}'],
        [/{@await /gim, '{#await '],
        [/{@then /gim, '{:then '],
        [/{@catch /gim, '{:catch '],
        [/{@endawait}/gim, '{/await}'],
        [/{@key /gim, '{#key '],
        [/{@endkey}/gim, '{/key}'],
        [/{@changed /gim, '{#key '],
        [/{@endchanged}/gim, '{/key}'],
      ],
    }),
  ],
}
```

**Comment by sup-ham** [comment]
I also tried 
- using this base template https://github.com/roxiness/routify-starter
- add only sweetalert2

App.svelte
```
<script>
  import { Router } from "@roxi/routify";
  import { routes } from "../.routify/routes";
  import Swal from 'sweetalert2';

  console.info({Swal})
</script>

<style  global>
  @import "../assets/global.css";
</style>

<Router {routes} />
```

**Comment by PepsRyuu** [comment]
Looks like a circular issue. I need to improve the circular dependency resolution, but it's a fairly big task. There is a compiled version of the library in the package so you can use an ```@rollup/plugin-alias``` as a workaround. Really though, the library author should not be pointing ```module``` in their ```package.json``` to source code and instead should be pointing to a compiled ESM version of the module.

**Comment by sup-ham** [comment]
I dont know whether there is compiled ESM version. I can't  import directly from dist, it same. currently my workaround  is  by <script>  tag.

**Comment by PepsRyuu** [comment]
Fixed in ```0.18.0```.
